x/commitment/keeper/msg_server_cancel_vest.go

```diff
@@ -51,6 +51,20 @@ func (k msgServer) CancelVest(goCtx context.Context, msg *types.MsgCancelVest) (
 		return nil, sdkerrors.Wrapf(types.ErrInsufficientVestingTokens, "denom: %s, amount: %s", msg.Denom, msg.Amount)
 	}
 
+	// Update the uncommitted tokens amount
```

Comment:
@AustinoBombino just a recommendation, instead of manually incrementing the `i` counter in the `else` branches, one way you can approach that is to have the standard `i++` in the `for` expression and then have a `i--` within the `if vesting.TotalAmount.IsZero() {` branch that would make code visibility easier to understand I believe.